File: balloon/zeromq/src/test/python/publisher.py
__author__ = 'timoveldt'

import zmq
import time


def main():
    context = zmq.Context()

    pub = context.socket(zmq.PUB)
    pub.connect("tcp://localhost:5560")
    for n in range(1000):
        for topic in "ABC":
            msg = [topic, str(n)]
            pub.send_multipart(msg)
        time.sleep(0.25)
        print "Sent %d to topics" % n
    context.term()


if __name__ == "__main__":
    main()

File: balloon/zeromq/src/test/cpp/publisher.cpp
//
// Created by timoveldt on 6/26/15.
//


#include "zmq.hpp"

#include <iostream>
#include <sstream>
#include <unistd.h>

int main(int argc, char *argv[]) {
    zmq::context_t context(1);

    //  First, connect our subscriber socket
    zmq::socket_t publisher(context, ZMQ_PUB);
    publisher.connect("tcp://localhost:5560");

    const std::string topic = "CPP_TOPIC";
    for(int i = 5000; i< 6000; i++) {
        std::stringstream iss;

        publisher.send(topic.data(), topic.size(), ZMQ_SNDMORE);

        iss << i;
        const std::string message = iss.str();

        publisher.send(message.data(), message.size(), 0);

        iss.flush();

        usleep(250 * 100);
    }


    return 0;
}
